django/db/models/fields/__init__.py

```diff
@@ -379,6 +387,36 @@ def _check_choices(self):
             )
         ]
 
+    def _check_db_default(self, databases=None, **kwargs):
+        if self.db_default is NOT_PROVIDED or databases is None:
+            return []
+        errors = []
+        for db in databases:
+            if not router.allow_migrate_model(db, self.model):
+                continue
+
+            if not self.db_default.allowed_default:
+                error = (
+                    f"{self.db_default} cannot be used as"
+                    + f' `db_default` on database "{db}".'
+                )
+                errors.append(checks.Error(error, obj=self, id="fields.E012"))
+
```

Comment:
Small concern: could we check if for a database expression class, not instance, as well? I'm thinking that this looks right, and the later failure could be hard to debug:

```python
models.DateTimeField(db_default=Now)
```

Or we could defer that and leave it for type-checking to spot.